numpy/ma/core.pyi

```diff
@@ -455,7 +462,68 @@ class MaskedIterator:
 
 class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     __array_priority__: Any
-    def __new__(cls, data=..., mask=..., dtype=..., copy=..., subok=..., ndmin=..., fill_value=..., keep_mask=..., hard_mask=..., shrink=..., order=...): ...
+    @overload
+    def __new__(
+        cls,
+        data: _ArrayLike[_ScalarT],
+        mask: _ArrayLikeBool_co = nomask,
```

Comment:
maybe I'm wrong here, but I thought integer masks also worked

Reply:
This is the same annotation as in `__setmask__`

https://github.com/numpy/numpy/blob/8b2321d03c1abce671a15f9d08a97249bd2d161d/numpy/ma/core.pyi#L510

not sure if passing integer masks should be encouraged?

Reply:
> not sure if passing integer masks should be encouraged?

No idea to be honest. You probably know more about `np.ma` then I do, so your call.

Reply:
😄 not necessarily

tbh I'd be ok with punting on it for the sake of this issue, and if it needs updating, it can be updated throughout the file